packages/compose/src/hooks/use-debounce/index.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * useDebounce hook
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Use to debounce a value that is changing too frequently.
+ *
+ * Example usage:
+ *
+ *	// An example using the useDebounce hook with useResizeObserver()
+ *  // to debounce a rapidly changing width value
+ *
+ *	const [ resizeListener, sizes ] = useResizeObserver();
+ *	const debouncedSize = useDebounce(sizes.width, 100);
+ *
+ *	useEffect( () => {
```

Comment:
I find the usage a little strange. I'd feel like I shouldn't need a separate `useEffect` hook to take advantage of `useDebounce`, and that it should be self-contained in how I provide a callback behavior.

I guess it's a bit unique in that we need to provide more than a single argument: (a) the callback, (b) the delay.

I think how you're using `sizes.width` in this example fits pretty well to the definition of `deps`.

Assuming three argument hooks is not outside the realm of convention, I might imagine a signature:

```
useDebounce( callback, delay, deps )
```

Reply:
Good suggestion. I reworked the function by passing in the callback it simplifies both the usage of the hook and the hook itself.